api/src/main/java/io/opentelemetry/api/trace/propagation/W3CTraceContextPropagator.java

```diff
@@ -0,0 +1,239 @@
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.api.trace.TraceStateBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Implementation of the W3C TraceContext propagation protocol. See <a
+ * href="https://www.w3.org/TR/trace-context-1/">W3C Trace Context</a>.
+ *
+ * <p>This is the default propagator for {@link SpanContext}s. The {@link SpanContext} type is
+ * designed to support all the data propagated via W3C propagation natively.
+ */
+@Immutable
+public final class W3CTraceContextPropagator {
+  private static final Logger logger = Logger.getLogger(W3CTraceContextPropagator.class.getName());
+
+  private static final TraceState TRACE_STATE_DEFAULT = TraceState.builder().build();
+  static final String TRACE_PARENT = "traceparent";
+  static final String TRACE_STATE = "tracestate";
+  private static final List<String> FIELDS =
+      Collections.unmodifiableList(Arrays.asList(TRACE_PARENT, TRACE_STATE));
+
+  private static final String VERSION = "00";
+  private static final int VERSION_SIZE = 2;
+  private static final char TRACEPARENT_DELIMITER = '-';
+  private static final int TRACEPARENT_DELIMITER_SIZE = 1;
+  private static final int TRACE_ID_HEX_SIZE = TraceId.getHexLength();
+  private static final int SPAN_ID_HEX_SIZE = SpanId.getHexLength();
+  private static final int TRACE_OPTION_HEX_SIZE = TraceFlags.getHexLength();
+  private static final int TRACE_ID_OFFSET = VERSION_SIZE + TRACEPARENT_DELIMITER_SIZE;
+  private static final int SPAN_ID_OFFSET =
+      TRACE_ID_OFFSET + TRACE_ID_HEX_SIZE + TRACEPARENT_DELIMITER_SIZE;
+  private static final int TRACE_OPTION_OFFSET =
+      SPAN_ID_OFFSET + SPAN_ID_HEX_SIZE + TRACEPARENT_DELIMITER_SIZE;
+  private static final int TRACEPARENT_HEADER_SIZE = TRACE_OPTION_OFFSET + TRACE_OPTION_HEX_SIZE;
+  private static final int TRACESTATE_MAX_SIZE = 512;
+  private static final int TRACESTATE_MAX_MEMBERS = 32;
+  private static final char TRACESTATE_KEY_VALUE_DELIMITER = '=';
+  private static final char TRACESTATE_ENTRY_DELIMITER = ',';
+  private static final Pattern TRACESTATE_ENTRY_DELIMITER_SPLIT_PATTERN =
+      Pattern.compile("[ \t]*" + TRACESTATE_ENTRY_DELIMITER + "[ \t]*");
+  private static final Set<String> VALID_VERSIONS;
+  private static final String VERSION_00 = "00";
+  private static final TextMapPropagator INSTANCE =
```

Comment:
Moving these tho the inner class, removing the interface implemenation and declaring getInstance as returning only `TextMapPropagator` instead of `W3CTraceContextPropagator` where the only changes here. Unfortunately it seems GitHub does not show that well.